### ISSUE TITLE
Any plan/guide to implement analytic pose gradient?

### ISSUE THREAD

**Issue by intMinsu** [hbb1/diff-surfel-rasterization#30]
Hi, thank you for this great work!
I am interested in both using 2D gaussian splatting for geometric accuracies and test-time pose regression w.r.t gaussian primitives.
I think the latter can be implemented as in [gsplat](https://github.com/nerfstudio-project/gsplat/blob/ddf88c6ac3e1890146f3dc0bcfae4b7d36b0e34a/examples/simple_trainer_2dgs.py#L333) written in python or in CUDA as in [Gaussian Splatting SLAM](https://github.com/rmurai0610/diff-gaussian-rasterization-w-pose). 
Presumably, I think CUDA version may show better performance, so I want to implement diff-surfel-rasterization-w-pose in CUDA.

Unfortunately, I have never done CUDA Programming, so I am not sure CUDA implementation for analytic pose gradient w.r.t 2D Gaussian primitives can be achieved without hassle.

Can you look out the [git diff](https://github.com/rmurai0610/diff-gaussian-rasterization-w-pose/commit/8fa512eae9228d08d88058c588caf25032b7d142) briefly and give any advices to combine two works?

Best regards

**Comment by hbb1** [comment]
You can start with this mode that natively supports gradient flow to camera parameters.  Then, you may also use it to validate you customize cuda kernel.

[gaussian_renderer/__init__.py](https://github.com/hbb1/2d-gaussian-splatting/blob/main/gaussian_renderer/__init__.py#L64C1-L75C118):

```python
    if pipe.compute_cov3D_python:
        # currently don't support normal consistency loss if use precomputed covariance
        splat2world = pc.get_covariance(scaling_modifier)
        W, H = viewpoint_camera.image_width, viewpoint_camera.image_height
        near, far = viewpoint_camera.znear, viewpoint_camera.zfar
        ndc2pix = torch.tensor([
            [W / 2, 0, 0, (W-1) / 2],
            [0, H / 2, 0, (H-1) / 2],
            [0, 0, far-near, near],
            [0, 0, 0, 1]]).float().cuda().T
        world2pix =  viewpoint_camera.full_proj_transform @ ndc2pix
        cov3D_precomp = (splat2world[:, [0,1,3]] @ world2pix[:,[0,1,3]]).permute(0,2,1).reshape(-1, 9) # column major
```

**Comment by intMinsu** [comment]
Thank you for prompt answer! I will implement this soon!

**Comment by hbb1** [comment]
@intMinsu 
The `compute_cov3D_python ` is the most convenient.

If you need further acceleration, here is a [referenced CUDA implementation](https://github.com/Dwawayu/diff-surfel-rasterization) from my labmate.  It works well with extrinsics  but bases on an older version (40% slow than the new code-base). You may need some adaptions to the existing codebase. 

Welcome to submit a PR if you would like to contribute and need some double checks!

**Comment by intMinsu** [comment]
Thank you for providing implementation! 😃😃😃
